### DISCUSSION TITLE
EEPROM I2C Connection Fail using flowdeck

### DISCUSSION THREAD

**Discussion by postmaster87** [bitcraze/discussions#409] [Q&A]
Hello,

I've looked at the other posts regarding this error but haven't seen anything like I'm experiencing. I have a CF 2.1 that works fine without the Flow V2 attached. After connecting a flow deck it fails the self test and shows EEPROM I2C connection [Fail], SYS: configblock [Fail], etc... in the console output (see attached). The issue does not appear on other CF's with the same deck attached so I'm thinking there's a connection/hw issue with the CF itself. I've tried other working decks on the CF in question but all cause the self test failure. I've also reflashed the firmware but still having the self test fail. Please advise - thanks!
![cfclientConsoleFlowdeck](https://user-images.githubusercontent.com/17578494/202306206-82095129-954b-445d-bd5e-8356cf599ef2.PNG)

**Comment by tobbeanton** [comment]
Most likely there is a glitch in the connector on the crazyflie board. Your can try to fix it using a needle:

1. Get a small needle (around 0.5mm in diameter)
2. Start with one pin and push the needle though the contact. If there is no or minimal friction it needs to be fixed.
3. Use the tip of the needle to get behind the inner golden "tongue" and gently bend it out. There are two tongues, front and back, that can be adjusted.
4. Test the contact again. If it was bent out to much test from the bottom to not risking over-bending the tongue the wrong way. If it was bent to little, redo it.
5. Repeat this for all contacts.

Also check the soldering of the connectors visually. You can take the needle and poke the soldered legs a little bit and see if it moves. It shouldn't move.

**Comment by postmaster87** [comment]
Thanks for the quick reply. I do not have a tool small enough to pry at the header connectors but I did run a standard male header through all the connectors and the resistance felt the same. Also when looking at them through a magnifier they all seem identical. The solder joints look fine and nothing is loose. Do you know which pins could be causing this issue? I can go reflow those joints and probably find something to work on the connectors if I know which one(s) could be at fault. Thanks

**Reply by knmcguire** [reply]
There is an expansionport picture with the pinout to be found here in the documentation: https://www.bitcraze.io/documentation/system/platform/cf2-expansiondecks/#expansion-port-pinout. Perhaps reflow the I2C first, as that is what the error is giving on the console?
